geopandas/geodataframe.py

```diff
@@ -1470,6 +1482,11 @@ def __finalize__(self, other, method=None, **kwargs):
                     f"Please ensure this column from the first DataFrame is not "
                     f"repeated."
                 )
+        elif method == "unstack":
+            # unstack adds multiindex columns and reshapes data.
+            # it never makes sense to retain geometry column
+            self._geometry_column_name = DEFAULT_GEO_COLUMN_NAME
```

Comment:
Should we set this to None then?

Reply:
I wondered about that as well - I went with this to be "consistent" with if you'd constructed something using the ordinary geodataframe init constructor. (It also happens to be consistent with prior to this PR). But I think there is a good argument this is a bug - and if not still progress in the right direction, so would be happy to change to None.
```suggestion
            self._geometry_column_name = None
```

Reply:
Let's give None a try here. I committed your suggestion